runtime/debug/block.go

```diff
@@ -0,0 +1,30 @@
+package debug
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/prysmaticlabs/prysm/v5/config/features"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/interfaces"
+	"github.com/prysmaticlabs/prysm/v5/io/file"
+	log "github.com/sirupsen/logrus"
+)
+
+// SaveBlockToTemp saves block to temp directory in ssz format.
+func SaveBlockToTemp(block interfaces.ReadOnlySignedBeaconBlock) {
+	if !features.Get().SaveInvalidBlock && !features.Get().SaveUnavailableBlock {
+		return
+	}
+	filename := fmt.Sprintf("beacon_block_%d.ssz", block.Block().Slot())
+	fp := path.Join(os.TempDir(), filename)
+	log.Warnf("Writing block to disk at %s", fp)
+	enc, err := block.MarshalSSZ()
+	if err != nil {
+		log.WithError(err).Error("Failed to ssz encode block")
+		return
+	}
+	if err := file.WriteFile(fp, enc); err != nil {
```

Comment:
wonder if a write file like this should be in a separate go routine to not block , maybe it doesn't matter that much since you need the feature turned on anyways